pkg/updater/updater.go

```diff
@@ -0,0 +1,224 @@
+package updater
+
+import (
+	"fmt"
+	"github.com/go-git/go-billy/v5"
+	"github.com/go-git/go-billy/v5/memfs"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/go-git/go-git/v5/plumbing/transport/ssh"
+	"github.com/go-git/go-git/v5/storage/memory"
+	"github.com/rs/zerolog/log"
+	"gopkg.in/yaml.v2"
+	"io"
+	"os"
+	"time"
+)
+
+var (
+	sshKeyPath    = os.Getenv("SSH_KEY_PATH")
+	sshKeyPass    = os.Getenv("SSH_KEY_PASS")
+	sshCommitUser = os.Getenv("SSH_COMMIT_USER")
+	sshCommitMail = os.Getenv("SSH_COMMIT_MAIL")
+)
+
+type ArgoOverrideFile struct {
+	Helm struct {
+		Parameters []ArgoParameterOverride `yaml:"parameters"`
+	} `yaml:"helm"`
+}
+
+type ArgoParameterOverride struct {
+	Name        string `yaml:"name"`
+	Value       string `yaml:"value"`
+	ForceString bool   `yaml:"forceString"`
+}
+
+type GitRepo struct {
+	RepoURL    string
+	BranchName string
+	Path       string
+	fs         billy.Filesystem
+	localRepo  *git.Repository
+	sshAuth    *ssh.PublicKeys
+}
+
+func (repo *GitRepo) Clone() error {
+	var err error
+
+	repo.fs = memfs.New()
```

Comment:
Would it make sense to use actual file system in the future, so that we don't fetch the repository on every deployment

Reply:
I'm inclined towards the in-memory storage approach. Given that we're only fetching a single branch, the impact should be minimal unless we're dealing with an exceptionally large repository — which presents its own set of challenges.

Moreover, with potentially dozens of repositories varying by setup, cache management could become cumbersome.

However, I acknowledge that the current approach may not seem optimal. I'm still weighing the best path forward.